pkg/kubelet/kubelet_pods.go

```diff
@@ -736,12 +736,8 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 // OkToDeletePod returns true if all required node-level resources that a pod was consuming have
 // been reclaimed by the kubelet.  Reclaiming resources is a prerequisite to deleting a pod from the
 // API server.
-func (kl *Kubelet) OkToDeletePod(pod *v1.Pod) bool {
-	if pod.DeletionTimestamp == nil {
```

Comment:
An evicted pod's timestamp is never set.  I moved that check into the status manager.